### PR TITLE
feat!: make CLI more ergonomic

### PR DESCRIPTION
**Pull request by vhyrro**
stacked on #31

This PR aims to refactor the CLI to be generally easier to use. Note: this is a breaking change and breaks compatibility with luarocks itself.

### CODE REVIEW

**Comment by vhyrro** [comment]
woops, have to fix the commit history

**Comment by vhyrro** [comment]
These are all the breakages worth making at this point in time I believe :p

File: rocks-bin/src/list.rs [hunk 1]
```diff
@@ -6,8 +6,6 @@ use text_trees::{FormatCharacters, StringTreeNode, TreeFormatting};
 
 #[derive(Args)]
 pub struct ListCmd {
-    #[arg(long)]
-    outdated: bool,
```
Comment (mrcjkb) [inline rocks-bin/src/list.rs:10]: question: this is something we use in rocks.nvim.
Is there a plan for an equivalent?

**Review by mrcjkb — APPROVED** [review]

File: rocks-bin/src/list.rs (hunk 1 above)
Comment (vhyrro) [reply rocks-bin/src/list.rs:10]: Yeah absolutely. I've been considering putting this under `rocks update --dry-run` or `rocks update --list-only`, but other ideas are welcome. Having this under "list" didn't make much sense to me.

Comment (vhyrro) [reply rocks-bin/src/list.rs:10]: Implemented in the form of a `rocks update --list` subcommand. I think that fits the bill very nicely. Happy to hear thoughts on that.

Comment (mrcjkb) [reply rocks-bin/src/list.rs:10]: I'm not sure how I feel about it being an `update --list` flag. 
npm, yarn and ruby gems all have an `outdated` subcommand. pip has a `list --outdated` flag.

An `update --dry-run` flag makes more sense to me, as the name of the flag implies that the update won't be applied.

Comment (vhyrro) [reply rocks-bin/src/list.rs:10]: The thing with `--dry-run` is that it doesn't hint at the fact that it'll print machine-readable output about updates. I think that having a dedicated `outdated` command is nice and descriptive. Maybe even `rocks debug outdated`? `list --outdated` is not very discoverable, I only discovered it was a thing when you told me a while back :p

Comment (mrcjkb) [reply rocks-bin/src/list.rs:10]: `outdated` sounds good to me.

**Pull request merged**